core/src/main/kotlin/fr/sncf/osrd/standalone_sim/SafetySpeed.kt

```diff
@@ -49,12 +51,32 @@ fun makeSafetySpeedRanges(
                     it.receptionSignal == RJSTrainStop.RJSReceptionSignal.SHORT_SLIP_STOP
                 )
             }
+            .filter {
+                // Stops that are in a range of ETCS_LEVEL2 signaling system are handled later
+                !isStopInSignalingSystemRange(
+                    it.offset,
+                    signalingRanges,
+                    ETCS_LEVEL2.id,
+                )
+            }
             .toMutableList()
-    makeEndOfPathStop(rawInfra, routes, signalOffsets)?.let { stopsWithSafetySpeed.add(it) }
+
+    // Last buffer stop is handled separately when in a ETCS_LEVEL2 signaling system range
+    if (
+        schedule.isNotEmpty() &&
+            !isStopInSignalingSystemRange(
+                schedule.last().pathOffset,
+                signalingRanges,
+                ETCS_LEVEL2.id,
+            )
+    ) {
+        makeEndOfPathStop(rawInfra, routes, signalOffsets)?.let { stopsWithSafetySpeed.add(it) }
+    }
```

Comment:
This block could be simplified if we first build a list with all `SafetySpeedStop` (including the last one), and *then* filter out the ones in ETCS ranges